backend/test/customerTests/service.test.ts

```diff
@@ -0,0 +1,75 @@
+import * as sinon from 'sinon';
+import sinonStubPromise from 'sinon-stub-promise';
+import { expect } from 'chai';
+import 'mocha';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import { config } from '../../config';
+
+import CustomerService from '../../src/Customer/customer.service';
+import CustomerModel from '../../src/Customer/customer.models';
+import emailService from '../../src/Email/email.service';
+
+sinonStubPromise(sinon);
+
+let sandbox: sinon.SinonSandbox;
+
+const mockCustomer = {
+    name: 'john',
+    email: 'doe'
+};
+
+const mockUserFromDb = {
+    userId: 1,
+    name: 'john',
+    email: 'doe',
+    role: 'admin',
+    password: bcrypt.hashSync('password', config.bcrypt_salt),
+    resetPasswordToken: null,
+    resetPasswordExpires: null
+};
+
+describe('Customer Service Test', () => {
+    beforeEach(() => {
+        sandbox = sinon.createSandbox();
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    it('Test get all customers', async () => {
+        const customerService = new CustomerService();
+        sandbox.stub(CustomerModel, 'getAll').resolves('foo');
+        const res = await customerService.getAllCustomers();
+        expect(res).to.equal('foo');
+    });
+
+    it('Test create new customer', async () => {
```

Comment:
Same as before test for an error when creating a new customer that already exists

Reply:
We don't make customers unique the same way we do for the users and so we won't have a test like that for customers.